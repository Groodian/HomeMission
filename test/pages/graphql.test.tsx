import { gql, InMemoryCache } from '@apollo/client';
import React from 'react';
import GraphQLTest from '../../src/pages/graphql';
import { render } from '../testUtils';

jest.mock('next/router', () => require('next-router-mock'));

const apolloCache = new InMemoryCache();
apolloCache.writeQuery({
  query: gql`
    query User {
      user {
        id
        name
        status
      }
    }
  `,
  data: {
    user: {
      __typename: 'User',
      id: 'Baa',
      name: 'Baa',
      status: 'Healthy',
    },
  },
});

describe('GraphQL test page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<GraphQLTest />, { apolloCache });
    expect(asFragment()).toMatchSnapshot();
  });
});
