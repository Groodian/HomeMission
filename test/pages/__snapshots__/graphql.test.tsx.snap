// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphQL test page matches snapshot 1`] = `
<DocumentFragment>
  <div>
    You're signed in as Baa and you're Healthy.
    <div>
      <input
        placeholder="your new name..."
        type="text"
      />
      <input
        type="button"
        value="change"
      />
    </div>
  </div>
</DocumentFragment>
`;
